### PR TITLE
New Serverless Pattern - ALB to ECS (Spring Boot Java 21)

### PR DESCRIPTION
**Pull request by biswanathmukherjee**
*Issue #, if available:*

*Description of changes:*
This sample project deploys a Java 21 Spring Boot application on Amazon ECS Fargate with Application Load Balancer to route traffic between two ECS Tasks.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: alb-ecs-java-sam/create-books-aws.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+-d '{
+"title": "The Great Gatsby3",
+"author": "F. Scott Fitzgerald3",
+"pageCount": 180,
+"category": "Fiction"
+}'
+
+
+
+
```
Comment (parikhudit) [inline alb-ecs-java-sam/create-books-aws.sh:58]: Extra lines, please remove

File: alb-ecs-java-sam/create-books-local.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+-d '{
+"title": "The Great Gatsby3",
+"author": "F. Scott Fitzgerald3",
+"pageCount": 180,
+"category": "Fiction"
+}'
+
+
+
+
```
Comment (parikhudit) [inline alb-ecs-java-sam/create-books-local.sh:55]: Please remove extra lines

File: alb-ecs-java-sam/src/main/java/com/aws/example/controller/BookManagementController.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+		return returnValue;
+	}
+
+	@GetMapping("/container-ip")
+	public String getContainerPrivateIp() {
+		String privateIp;
+		HttpClient client = HttpClient.newHttpClient();
+		HttpRequest request = HttpRequest.newBuilder()
+										.uri(URI.create(metadataUri))
```
Comment (parikhudit) [inline alb-ecs-java-sam/src/main/java/com/aws/example/controller/BookManagementController.java:137]: Indentation issue

**Comment by parikhudit** [comment]
@biswanathmukherjee Can you please review all the files and remove extra lines between & at the end of file. Please also check all the files for indentation - its inconsistent.

File: alb-ecs-java-sam/create-books-aws.sh (hunk 1 above)
Comment (biswanathmukherjee) [reply alb-ecs-java-sam/create-books-aws.sh]: Fixed

File: alb-ecs-java-sam/create-books-local.sh (hunk 2 above)
Comment (biswanathmukherjee) [reply alb-ecs-java-sam/create-books-local.sh]: Fixed.

File: alb-ecs-java-sam/template.yaml [hunk 4]
```diff
@@ -0,0 +1,248 @@
+AWSTemplateFormatVersion: 2010-09-09
+Transform: AWS::Serverless-2016-10-31
+Description: This template deploys ALB with ECS Farget cluster to deploy a containerised Spring Boot Application
```
[the remark this answers is not in the document]
Comment (biswanathmukherjee) [reply alb-ecs-java-sam/template.yaml]: Fixed

File: alb-ecs-java-sam/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
+{
+  "title": "Deploying Spring-boot application on Amazon ECS Fargate with ALB configuration",
+  "description": "This sample project deploys a Java 21 Spring Boot application on Amazon ECS Fargate with Application Load Balancer to route traffic between two ECS Tasks.",
```
[the remark this answers is not in the document]
Comment (biswanathmukherjee) [reply alb-ecs-java-sam/example-pattern.json]: Fixed.

**Pull request merged**